File: mikado-method/test-data-builders-kata/typescript/test/com/murex/tbw/report/EducationalBookBuilder.ts
import { Author, Category, EducationalBook } from "../../../../../src/com/murex/tbw/domain/book";
import { Language } from "../../../../../src/com/murex/tbw/domain/country";
import { AuthorBuilder } from "./AuthorBuilder";

export class EducationalBookBuilder {
    private author: Author = AuthorBuilder.anAuthor().build();
    private title: string = "Martine fait des tests avec Caro et ses copains";
    private category: Category = Category.COMPUTER;
    private cost: number = 1000;
    private language: Language = Language.English;

    build(): EducationalBook {
        return new EducationalBook(this.title, this.cost, this.author, this.language, this.category);
    }

    categorizedAs(category: Category): EducationalBookBuilder {
        this.category = category;
        return this;
    }

    costing(cost: number) {
        this.cost = cost;
        return this;
    }

    titled(title: string): EducationalBookBuilder {
        this.title = title;
        return this;
    }

    writtenBy(author: Author): EducationalBookBuilder {
        this.author = author;
        return this;
    }

    writtenIn(language: Language): EducationalBookBuilder {
        this.language = language;
        return this;
    }

    static aBook(): EducationalBookBuilder {
        return new EducationalBookBuilder();
    }
}

File: mikado-method/test-data-builders-kata/csharp/BookInvoicing.Tests/EducationalBookBuilder.cs
using BookInvoicing.Domain.Book;
using BookInvoicing.Domain.Country;

namespace BookInvoicing.Tests
{
    public class EducationalBookBuilder
    {
        private string _name = "myName";
        private int _price = 25;
        private Author _author = AuthorBuilder.AnAuthor().Build();
        private Language _language = Language.English;
        private Category _category = Category.Computer;

        public static EducationalBookBuilder AnEducationalBook()
        {
            return new EducationalBookBuilder();
        }

        public EducationalBookBuilder Named(string name)  
        {
            _name = name;
            return this;
        }

        public EducationalBookBuilder Costing(int price)
        {
            _price = price;
            return this;
        }

        public EducationalBookBuilder WrittenBy(Author author)
        {
            _author = author;
            return this;
        }

        public EducationalBookBuilder WrittenIn(Language language)  
        {
            _language = language;
            return this;
        }

        public EducationalBookBuilder CategorizedIn(Category category)
        {
            _category = category;
            return this;
        }

        public EducationalBook Build()
        {
            return new EducationalBook(_name, _price, _author, _language, _category);
        }
    }
}